eZ/Publish/Core/QueryType/BuildIn/AbstractQueryType.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\QueryType\BuildIn;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeIdentifier;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalAnd;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Subtree;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Visibility;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\QueryType\OptionsResolverBasedQueryType;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+abstract class AbstractQueryType extends OptionsResolverBasedQueryType
+{
+    private const SORT_CLAUSE_NAMESPACE = '\eZ\Publish\API\Repository\Values\Content\Query\SortClause\\';
+
+    public const DEFAULT_LIMIT = 25;
+
+    /** @var \eZ\Publish\API\Repository\Repository */
+    protected $repository;
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    protected $configResolver;
+
+    public function __construct(Repository $repository, ConfigResolverInterface $configResolver)
+    {
+        $this->repository = $repository;
+        $this->configResolver = $configResolver;
+    }
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'filter' => static function (OptionsResolver $resolver): void {
+                $resolver->setDefaults([
+                    'content_type' => [],
+                    'visible_only' => true,
+                ]);
+
+                $resolver->setAllowedTypes('content_type', 'array');
+                $resolver->setAllowedTypes('visible_only', 'bool');
```

Comment:
Does `visible_only` have a default value ?

Reply:
Yes, default value is `true`. See https://github.com/ezsystems/ezpublish-kernel/blob/daabc4831ca10165a3d4238096b2a3c03f41aaa7/eZ/Publish/Core/QueryType/BuildIn/AbstractQueryType.php#L47